spec/unit/primitive_after_spec.rb

```diff
@@ -0,0 +1,103 @@
+require_relative '../../libraries/primitive_after'
+
+describe Choregraphie::After do
+  let(:block) { proc {} }
+  let(:choregraphie) do
+    Choregraphie::Choregraphie.new('test') do
+      after(&block)
+    end
+  end
+
+  describe 'standard behavior' do
+    before do
+      # all file manipulation are no-op
+      allow(FileUtils).to receive(:touch)
+      allow(FileUtils).to receive(:rm)
+    end
+
+    context 'when node has been installed long time ago' do
+      before do
+        allow(File).to receive(:exist?).with(/installed/).and_return(true)
+      end
+
+      context 'when a choregraphie is in progress' do
+        it 'must call its block' do
+          allow(File).to receive(:exist?).with(/inprogress/).and_return(true)
+          expect(block).to receive(:call)
+          choregraphie.cleanup.each(&:call)
+        end
+
+        it 'must mark its in progress state' do
+          expect(File).to receive(:exist?).with(/inprogress/).at_least(:once).and_return(false)
+          expect(FileUtils).to receive(:touch).with(/inprogress/)
+          choregraphie.before.each(&:call)
+        end
```

Comment:
this is kind of misleading: the `before` block is rather called in a "when we enter a choregraphie" context

Reply:
not sure to understand what you mean